frontend-v2/src/features/data/LoadDataStepper.tsx

```diff
@@ -289,23 +300,32 @@ const LoadDataStepper: FC<IStepper> = ({ csv = "", onCancel, onFinish }) => {
           paddingBottom: 1,
         }}
       >
-        {stepState.activeStep === 0 ? (
-          <Button onClick={onCancel}>Cancel</Button>
-        ) : (
-          <Button onClick={handleBack}>Back</Button>
-        )}
-        <Button
-          disabled={data.length === 0 || isFinished || errors.length > 0}
-          variant="contained"
-          color="primary"
-          onClick={
-            stepState.activeStep === stepLabels.length - 1
-              ? handleFinish
-              : handleNext
-          }
+        <Tooltip title='You can cancel and upload new file' placement='top' arrow>
+          <Button variant='outlined' onClick={onResetDataset} disabled={!data.length}>Cancel</Button>
```

Comment:
Maybe change this to call `onCancel` if there's no upload to reset?
```suggestion
          <Button variant='outlined' onClick={data.length === 0 ? onCancel : onResetDataset} disabled={!data.length}>Cancel</Button>
```

Reply:
Cannot use data.length, as data array is also populated when editing a dataset, but I have noticed that 'csv' prop is only passed when editing dataset, so I used that.

Reply:
👍